### PR TITLE
Pngextractiondev

### PR DESCRIPTION
**Pull request by Pavan-Bellam**
Supporting extraction of private tags.

### CODE REVIEW

**Comment by pradeeban** [comment]
I endorse this pull request. I will test tomorrow.

**Comment by Pavan-Bellam** [comment]
Thank you

File: modules/png-extraction/ImageExtractor.py [hunk 1]
```diff
@@ -102,19 +106,36 @@ def get_tuples(plan, outlist = None, key = ""):
...
+    headers=[]
+    if(publicheadersonly):
+        for aa in plan.dir():
+            headers.append(plan[aa])
+    else:
+        feature_list = open("featureset.txt").read().splitlines()
+        if len(feature_list)==0:
+            headers=[i for i in plan]
+        else:
+            headers.append(plan['PatientID'])
```
[the remark this answers is not in the document]
Comment (pradeeban) [reply modules/png-extraction/ImageExtractor.py:118]: KeyError: (0028, 0004) indicates, the image does not have the attribute "Photometric Interpretation"

https://dicom.innolitics.com/ciods/rt-dose/image-pixel/00280004

**Comment by pradeeban** [comment]
When there is no "featureset.txt" present while "        "PublicHeadersOnly": false, it throws the below error.


python3 ImageExtractor.py
/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/pydicom/filereader.py:300: UserWarning: Expected explicit VR, but found implicit VR - using implicit VR for reading
  warnings.warn(message, UserWarning)
Traceback (most recent call last):
  File "ImageExtractor.py", line 522, in <module>
    initialize_config_and_execute(args)
  File "ImageExtractor.py", line 99, in initialize_config_and_execute
    failed, maps_directory, meta_directory, LOG_FILENAME, metadata_col_freq_threshold, t_start)
  File "ImageExtractor.py", line 398, in execute
    feature_list = open("featureset.txt").read().splitlines()
FileNotFoundError: [Errno 2] No such file or directory: 'featureset.txt'



Can you please make a try-catch that will just get all the tags when featureset.txt is not found? I thought that is what you are doing from the README.

**Comment by pradeeban** [comment]
Also, I have left another comment regarding a Key Error in the code directly.

**Comment by pradeeban** [comment]
Hi,
#258 is now fixed in the dev branch. But your pull request goes beyond that, also attempting to implement #282.

If you do a "git pull" from the dev branch and adapt your changes accordingly, with your existing progress, I believe you could easily implement #282.

Please make sure to keep a note of all your pull requests and contributions in your project proposal (even if they are not merged). Because these help highlight your strength in Python and DICOM.

**Pull request closed without merging**